go.mod

```diff
@@ -4,31 +4,49 @@ go 1.12
 
 require (
 	cloud.google.com/go v0.34.0
+	github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20191108070603-f326d82a7d0f
 	github.com/aws/aws-sdk-go v1.21.6
 	github.com/blang/semver v3.5.1+incompatible
+	github.com/cockroachdb/cmux v0.0.0-20170110192607-30d10be49292 // indirect
+	github.com/coreos/bbolt v1.3.3 // indirect
 	github.com/coreos/etcd v3.2.18+incompatible
 	github.com/coreos/go-semver v0.2.0 // indirect
+	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f // indirect
 	github.com/denverdino/aliyungo v0.0.0-20180409134525-ed76a3738861 // indirect
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
 	github.com/digitalocean/godo v1.19.0
 	github.com/go-ini/ini v1.32.0 // indirect
+	github.com/gogo/protobuf v1.3.1 // indirect
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	github.com/golang/protobuf v1.2.0
+	github.com/golang/groupcache v0.0.0-20191027212112-611e8accdfc9 // indirect
+	github.com/golang/protobuf v1.3.2
+	github.com/google/btree v1.0.0 // indirect
 	github.com/gophercloud/gophercloud v0.0.0-20190216224116-dcc6e84aef1b
+	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
+	github.com/grpc-ecosystem/grpc-gateway v1.12.1 // indirect
+	github.com/jonboulle/clockwork v0.1.0 // indirect
 	github.com/kr/fs v0.0.0-20131111012553-2788f0dbd169 // indirect
-	github.com/pkg/errors v0.8.0 // indirect
+	github.com/modern-go/reflect2 v1.0.1 // indirect
 	github.com/pkg/sftp v0.0.0-20180127012644-738e088bbd93 // indirect
+	github.com/prometheus/client_golang v1.2.1 // indirect
+	github.com/spf13/afero v1.2.2 // indirect
 	github.com/ugorji/go v0.0.0-20170107133203-ded73eae5db7 // indirect
-	golang.org/x/net v0.0.0-20190108225652-1e06a53dbb7e
+	github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2 // indirect
+	go.etcd.io/bbolt v1.3.3 // indirect
+	golang.org/x/net v0.0.0-20191002035440-2ec189313ef0
 	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
 	google.golang.org/api v0.0.0-20180127000325-725ebdd8d555
-	google.golang.org/grpc v1.17.0
+	google.golang.org/grpc v1.24.0
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/warnings.v0 v0.1.2 // indirect
-	gopkg.in/yaml.v2 v2.2.2
+	gopkg.in/yaml.v2 v2.2.3
 	k8s.io/apimachinery v0.0.0-20180105065639-91d8586aac31
 	k8s.io/client-go v9.0.0+incompatible
 	k8s.io/klog v0.1.0
 	k8s.io/kops v0.0.0-20190513135614-24206f04f5eb
 	k8s.io/kubernetes v1.12.1
 	k8s.io/utils v0.0.0-20180208044234-258e2a2fa645
 )
+
+replace k8s.io/kops => github.com/bittopaz/kops v1.4.2-0.20191112124010-4d74437144d4
```

Comment:
@justinsb due to the really old version of the kops dependency, I have to replace it when testing locally.

Can you please show me a little bit on "how to update kops version in etcd-manger"? Thanks!

Reply:
You're right - it should be easy, but unfortunately some of the kubernetes go.mod files have dependency versions of v0.0.0, and thus it looks like we need to specify the versions of all those dependencies (even if we aren't using them, it seems).  Working on this in #278